### PR TITLE
[Validator] Add the divisibleBy option to the Count constraint

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | TODO

From my experience, it is sometimes useful to assert that the number of elements in a collection is a multiple of X.

### CODE REVIEW

**Review by OskarStark — APPROVED** [review]

File: src/Symfony/Component/Validator/Resources/translations/validators.fr.xlf [hunk 1]
```diff
@@ -370,6 +370,10 @@
                 <source>This value is not a valid hostname.</source>
                 <target>Cette valeur n'est pas un nom d'hôte valide.</target>
             </trans-unit>
+            <trans-unit id="96">
+                <source>This collection should contain a number of elements that is a multiple of {{ compared_value }}.</source>
+                <target>Cette collection doit contenir un nombre d'élements étant un multiple de {{ compared_value }}.</target>
```
Comment (atailouloute) [inline src/Symfony/Component/Validator/Resources/translations/validators.fr.xlf]:
```suggestion
                <target>Cette collection doit contenir un nombre d'éléments étant un multiple de {{ compared_value }}.</target>
```

File: src/Symfony/Component/Validator/Resources/translations/validators.fr.xlf [hunk 2]
```diff
@@ -370,6 +370,10 @@
                 <source>This value is not a valid hostname.</source>
                 <target>Cette valeur n'est pas un nom d'hôte valide.</target>
             </trans-unit>
+            <trans-unit id="96">
+                <source>This collection should contain a number of elements that is a multiple of {{ compared_value }}.</source>
+                <target>Cette collection doit contenir un nombre d'éléments étant un multiple de {{ compared_value }}.</target>
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Validator/Resources/translations/validators.fr.xlf]: `Cette collection doit contenir un nombre d'éléments multiple de {{ compared_value }}.`

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/Validator/Resources/translations/validators.fr.xlf (hunk 2 above)
Comment (fancyweb) [reply src/Symfony/Component/Validator/Resources/translations/validators.fr.xlf]: I changed the base English sentence to `The number of elements in this collection should be a multiple of {{ compared_value }}.` thanks to @ciaranmcnulty advice.

Comment (fancyweb) [reply src/Symfony/Component/Validator/Resources/translations/validators.fr.xlf]: Can you recheck the French translation please?

**Comment by HeahDude** [comment]
What issue does this fix considering we already have a dedicated contraint?

**Comment by fancyweb** [comment]
Currently you have to do everything manually:
```php
new Callback(static function ($collection, ExecutionContextInterface $context): void {
    $context
        ->getValidator()
        ->inContext($context)
        ->validate(\count($collection), new DivisibleBy([
            'value' => 3,
        ]));
}),
```

**Comment by HeahDude** [comment]
What about supporting countable in https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Validator/Constraints/DivisibleByValidator.php#L21 this instead?

**Comment by fancyweb** [comment]
I thought about this but `Count` is made for collections. Otherwise, we should also support countable for other comparisons consrraints such as `GreaterThan`, `LessThan`, etc.

**Comment by HeahDude** [comment]
Why not? :)

**Comment by fancyweb** [comment]
That implies way more work and it’s also less logical to use from a DX POV.
The collection « count » is divisible by X, not the collection itself. My
collection cannot be « greater than » X, it’s an array/object. Its « count
» must be minimum X. Etc. I’m sure the Count constraint was made for this.

**Comment by HeahDude** [comment]
>The collection « count » is divisible by X, not the collection itself. 

It is usually different, but when an object is countable, this is just the same thing.

**Comment by nicolas-grekas** [comment]
I think this approach (this PR) is more clear to me also.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @fancyweb.

**Pull request merged**